### DISCUSSION TITLE
Allow configuration of PrimeNG via a provider like `providePrimeNgConfig()`

### DISCUSSION THREAD

**Discussion by jreidgreer** [primefaces/community#2678] [PrimeNG] [answered]
I'm loving v18! One thing that feels a bit unusual from how most library configurations are set in Angular is requiring configuration in a component via `PrimeNGConfig`.

## Proposal:
I would propose a simple provider to pass in the configuration in the module:

**Current:**
```ts
@Component({
  selector: "app-root",
  templateUrl: "./app.component.html",
})
export class AppComponent implements OnInit {
  constructor(private primengConfig: PrimeNGConfig) {}

  ngOnInit() {
    this.primengConfig.theme.set({ preset: Aura });
    this.primengConfig.ripple.set(true);
  }
}
```

**Proposed:**
```ts
export const appConfig: ApplicationConfig = {
  providers: [
    provideRouter(routes),
    provideHttpClient(),
    providePrimeNgConfig({
      ripple: true,
      theme: { preset: Aura },
    }),
  ],
};
```

This makes the code more declarative, colocates application config better, and makes it more idiomatic to Angular.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the great idea! 

https://github.com/primefaces/primeng/issues/16716
